pkg/testcase/ingressdns.go

```diff
@@ -71,7 +71,11 @@ func TestDnsAccess(applyWorkload, deleteWorkload bool) {
 }
 
 func TestIngressRoute(applyWorkload, deleteWorkload bool, apiVersion string) {
-	publicIp := fmt.Sprintf("%s.nip.io", shared.FetchNodeExternalIP()[0])
+	workerNodes, err := shared.GetNodesByRoles("worker")
+	Expect(workerNodes).NotTo(BeEmpty())
+	Expect(err).NotTo(HaveOccurred())
+	publicIp := fmt.Sprintf("%s.nip.io", workerNodes[0].ExternalIP)
+
 	if applyWorkload {
```

Comment:
can we maintain the pattern and have the apply workload check first? because if the apply fail first its better

Reply:
ah for this one I needed to modify for this first because the publicIp is required during the apply, since there's a lot of file rewrite logic in there to move it from the template to the actual applied file with the IPs

Reply:
got you
tks!